app/services/tagging/common_ancestor_finder.rb

```diff
@@ -0,0 +1,54 @@
+module Tagging
+  class CommonAncestorFinder
+    def self.call
+      new.find_all
+    end
+
+    def find_all
+      content_enum = Services.rummager.search_enum({ fields: %w[content_id title] }, page_size: 100)
+      filtered_content_enum = content_enum.lazy.select { |c| c.has_key?('content_id') }
+      all_results = filtered_content_enum.map do |content_hash|
+        content_id = content_hash['content_id']
+        title = content_hash['title']
+        expanded_links_hash = Services.publishing_api.get_expanded_links(content_id).to_h
```

Comment:
At least for now, while most of the taxons are in draft, we'll need to use the `with_drafts` option to the expanded links endpoint to actually catch most of the tagging.

Reply:
the publishing api's default is to include drafts:
`  def get_expanded_links(content_id, with_drafts: true, generate: false)`

Reply:
Ah, ok, that's good :)